### ISSUE TITLE
io.downloads has to be documented and harmonized

### ISSUE THREAD

**Issue by cpaulik** [TUW-GEO/poets#60]
It seems to me that the three routines for downloading from `ftp` `sftp` and `http` all support slightly different things since the code is very different. So it should be documented what is supposed to be supported and we have to find a way to test the routines. At least the inner loops that go through the directory structure.

My suggestions would be to move this into its own package that is just responsible for getting the data from a remote source into a well defined directory structure on the local host. This new package could also be very useful for EODC. But something like that might already exist.

**Comment by tmistelbauer** [comment]
I agree that this should be a separate package. Back when I and @IsabellaP wrote this module, I did not find any package that provided the functions we needed. There might be now though.

**Comment by cpaulik** [comment]
I've written https://github.com/cpaulik/datedown for downloading GLDAS data but it should also cover most of the use cases of poets. Since it uses`wget` we don't have to bother too much with the details of the different protocols and it can download in parallel which should make it faster. If `wget` is not available also `curl` support could be envisioned. @IsabellaP I don't know if you plan to do some kind of automatic downloading for Sri Lanka but if so then you could look at this.
